[ews-build.webkit.org] Refuse to land security changes via Commit-Queue

JonWBedard opened the pull request:
#### f5c58155197797f60888a8c951fddfa3399a11cd
<pre>
[ews-build.webkit.org] Refuse to land security changes via Commit-Queue
<a href="https://bugs.webkit.org/show_bug.cgi?id=255559">https://bugs.webkit.org/show_bug.cgi?id=255559</a>
rdar://108166930

Reviewed by NOBODY (OOPS!).

* Tools/CISupport/ews-build/steps.py:
(ValidateChange.validate_bugzilla): Do not run Commit-Queue on sensitive changes.
* Tools/CISupport/ews-build/steps_unittest.py:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f5c58155197797f60888a8c951fddfa3399a11cd

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/3618 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/3659 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/3808 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/5044 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/3893 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/3592 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/23/builds/3759 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/3707 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/3139 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/3662 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/23/builds/3759 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/3261 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/4867 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/23/builds/3759 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/3232 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/3294 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/23/builds/3759 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/3291 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/4634 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/3671 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/2970 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/3578 "Passed tests") | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/30/builds/3205 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/3232 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/3237 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/412 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/3484 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -1749,6 +1749,10 @@ def run(self):
     @defer.inlineCallbacks
     def validate_bugzilla(self, patch_id):
+        if self.getProperty('sensitive', False) and self.getProperty('buildername', '').lower() == 'commit-queue':
+            rc = yield self.skip_build('Cannot land security changes with Commit-Queue')
```
1) Instead of marking the build as skipped, might be better to mark it failed.
2) It's better to remove cq+ flag and leave a comment on the bug, otherwise it might be hard to discover for the engineers.
3) Better to mention the solution as well, e.g.: `please use GitHub PR instead.`

Something like:
```
message = 'XYZ'
self.setProperty('build_finish_summary', message)
self.setProperty('comment_text', message)
self.build.addStepsAfterCurrentStep([LeaveComment(), SetCommitQueueMinusFlagOnPatch()])
```

Note that SetCommitQueueMinusFlagOnPatch will mark the build as failed (if `build_finish_summary` property is set).

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -1749,6 +1749,14 @@ def run(self):
+            self.setProperty('build_finish_summary', message)
+            self.setProperty('comment_text', message)
+            self.build.addStepsAfterCurrentStep([LeaveComment(), SetCommitQueueMinusFlagOnPatch()])
+            rc = yield self.fail_build(message)
```
This would fail the build immediately without a chance to run the steps added in above line. Might just do something like: `defer.returnValue(FAILURE)`

aj062 reviewed (commented):
r+ assuming this is tested.

JonWBedard committed:
[ews-build.webkit.org] Refuse to land security changes via Commit-Queue
https://bugs.webkit.org/show_bug.cgi?id=255559
rdar://108166930

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(ValidateChange.validate_bugzilla): Do not run Commit-Queue on sensitive changes.
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: https://commits.webkit.org/263093@main
